### PR TITLE
fix(NcDateTime): Respect language

### PR DESCRIPTION
**Pull request by Pytal**
### ☑️ Resolves

- Fix https://github.com/nextcloud/server/issues/15457

### 🖼️ Screenshots

🏚️ Before | 🏡 After
---|---
image | image

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [x] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: src/composables/useFormatDateTime.js [hunk 1]
```diff
@@ -58,14 +58,14 @@ export function useFormatDateTime(timestamp = Date.now(), opts = {}) {
 	const date = computed(() => new Date(unref(timestamp)))
 
 	const formattedFullTime = computed(() => {
-		const formatter = new Intl.DateTimeFormat(getCanonicalLocale(), wrappedOptions.value.format)
+		const formatter = new Intl.DateTimeFormat(getLanguage(), wrappedOptions.value.format)
```
Comment (susnux) [inline src/composables/useFormatDateTime.js:61]: This does not work, for example I have configured Nextcloud to use English (US) lanuage, but German locale (I want 24h system and dd.mm.yyyy days but English texts).
Now I receive 12h times and English date format.

**Review by susnux — CHANGES_REQUESTED** [review]
This does not work, for example I have configured Nextcloud to use English (US) lanuage, but German locale (I want 24h system and dd.mm.yyyy days but English texts).
Now also english

File: src/composables/useFormatDateTime.js (hunk 1 above)
Comment (susnux) [reply src/composables/useFormatDateTime.js:61]: So at least the full formatted time needs to use the users **locale** not language

File: src/composables/useFormatDateTime.js [hunk 2]
```diff
@@ -58,14 +58,14 @@ export function useFormatDateTime(timestamp = Date.now(), opts = {}) {
...
 		return formatter.format(date.value)
 	})
 
 	/** Time string formatted for main text */
 	const formattedTime = computed(() => {
 		if (wrappedOptions.value.relativeTime !== false) {
-			const formatter = new Intl.RelativeTimeFormat(getCanonicalLocale(), { numeric: 'auto', style: wrappedOptions.value.relativeTime })
+			const formatter = new Intl.RelativeTimeFormat(getLanguage(), { numeric: 'auto', style: wrappedOptions.value.relativeTime })
```
Comment (susnux) [inline src/composables/useFormatDateTime.js:68]: This is ok as this only is used for relative time which should be independent from **locale** and only depend on **lanuage**

**Review by susnux — APPROVED** [review]
:+1:

File: tests/unit/components/NcDateTime/NcDateTime.spec.js [hunk 3]
```diff
@@ -98,6 +98,31 @@ describe('NcDateTime.vue', () => {
...
+		beforeAll(() => {
+			// mock the language
+			document.documentElement.lang = 'de'
+		})
+		afterAll(() => {
+			// revert mock
+			document.documentElement.lang = 'en'
+		})
+
+		it('', () => {
```
Comment (ShGKme) [inline tests/unit/components/NcDateTime/NcDateTime.spec.js:111]: Test description is missing

File: tests/unit/components/NcDateTime/NcDateTime.spec.js [hunk 4]
```diff
@@ -98,6 +98,31 @@ describe('NcDateTime.vue', () => {
...
+			jest.setSystemTime(now)
+
+			const yesterday = Date.UTC(2023, 5, 23, 13, 59)
+			const wrapper = mount(NcDateTime, {
+				propsData: {
+					timestamp: yesterday,
+				},
+			})
+
+			expect(wrapper.element.textContent).toMatch('gestern')
```
Comment (ShGKme) [inline tests/unit/components/NcDateTime/NcDateTime.spec.js:122]:
```suggestion
			expect(wrapper.text()).toMatch('gestern')
```

**Review by ShGKme — APPROVED** [review]

**Comment by susnux** [comment]
@Pytal conflicts + maybe include Grigoriis comments :)

**Pull request merged**